test.py

```diff
@@ -114,25 +119,55 @@ def test_encoding(self):
             # Also, none of these should parse when interpreted as UTF-8
             self.assertRaises(Exception, str.decode, req.content, 'utf-8')
 
-
-class ContextManagerTest(unittest.TestCase):
-    '''Testing out using the context manager version'''
-
-    def test_context_manager(self):
-        '''Testing out using the context manager version'''
-        url = 'http://localhost:8080/basic/basic.asis'
-        # Shouldn't be able to fetch
-        self.assertRaises(requests.exceptions.ConnectionError,
-            requests.get, url)
-        with asis.Server('test', port=8080, server='gevent'):
-            # Make sure we can get requests now...
-            req = requests.get(url)
+    def test_charset_no_length(self):
+        '''When a charset is used, but a content-length is absent, none is provided.'''
+        # Just exercise this branch
```

Comment:
So, if no `Content-Length` header is provided in this case we don't want to create one. But normally if one is provided, then we adjust it according to its new compressed / encoded length.

However, it seems `bottle` is adding `Content-Length` anyways, so I don't know that there's really a meaningful assertion to be done here.